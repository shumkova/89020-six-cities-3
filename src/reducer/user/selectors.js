import NameSpace from "../name-space.js";

const NAME_SPACE = NameSpace.USER;


const getAuthorizationStatus = (state) => {
  return state[NAME_SPACE].authorizationStatus;
};

const getAuthInfo = (state) => {
  return state[NAME_SPACE].authInfo;
};

export {getAuthorizationStatus, getAuthInfo};

