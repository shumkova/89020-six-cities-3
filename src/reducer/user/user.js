import {extend} from "../../utils";
import EditAuthInfo from "../../adapters/edit-auth-info";
import {AppRoute} from "../../const";
import history from "../../history";

const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

const initialState = {
  authorizationStatus: AuthorizationStatus.NO_AUTH,
  authInfo: {},
};

const ActionTypes = {
  REQUIRED_AUTHORIZATION: `REQUIRED_AUTHORIZATION`,
  SET_INFO: `SET_INFO`,
};

const ActionCreator = {
  requireAuthorization: (status) => {
    return {
      type: ActionTypes.REQUIRED_AUTHORIZATION,
      payload: status,
    };
  },

  setUser: (data) => {
    return {
      type: ActionTypes.SET_INFO,
      payload: data,
    };
  }
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionTypes.REQUIRED_AUTHORIZATION:
      return extend(state, {
        authorizationStatus: action.payload,
      });

    case ActionTypes.SET_INFO:
      return extend(state, {
        authInfo: action.payload,
      });
  }

  return state;
};

const Operation = {
  checkAuth: () => (dispatch, getState, api) => {
    return api.get(`/login`)
      .then((response) => {
        dispatch(ActionCreator.requireAuthorization(AuthorizationStatus.AUTH));
        dispatch(ActionCreator.setUser(EditAuthInfo.parseUser(response.data)));
      })
      .catch((err) => {
        throw err;
      });
  },

  login: (authData) => (dispatch, getState, api) => {
    return api.post(`/login`, {
      email: authData.login,
      password: authData.password,
    })
      .then((response) => {
        const user = EditAuthInfo.parseUser(response.data);
        dispatch(ActionCreator.requireAuthorization(AuthorizationStatus.AUTH));
        dispatch(ActionCreator.setUser(user));
        history.push(AppRoute.ROOT);
      });
  }
};

export {
  ActionCreator,
  ActionTypes,
  AuthorizationStatus,
  Operation,
  reducer,
};
