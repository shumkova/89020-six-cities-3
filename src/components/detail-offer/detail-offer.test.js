import React from "react";
import renderer from "react-test-renderer";
import {AuthorizationStatus} from "../../reducer/user/user";
import DetailOffer from "./detail-offer.connect";
import history from "../../history";
import {Router} from "react-router-dom";
import NameSpace from "../../reducer/name-space";
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";

const mockStore = configureStore([]);

const HOTELS = [{
  bedrooms: 3,
  city: {
    location: {
      latitude: 52.370216,
      longitude: 4.895168,
      zoom: 10
    },
    name: `Amsterdam`
  },
  description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
  host: {
    avatar: `img/1.png`,
    id: 3,
    isPro: true,
    name: `Angelina`
  },
  id: 1,
  images: [`img/1.png`, `img/2.png`],
  isFavorite: false,
  isPremium: false,
  location: {
    latitude: 52.35514938496378,
    longitude: 4.673877537499948,
    zoom: 8
  },
  adults: 4,
  preview: `img/1.png`,
  price: 120,
  rating: 4.8,
  title: `Beautiful & luxurious studio at great location`,
  type: `apartment`
}];

const noop = () => {};

it(`Render DetailOffer`, () => {
  const store = mockStore({
    [NameSpace.DATA]: {
      hotels: HOTELS,
      applicationIsReady: true,
    },
    [NameSpace.APP]: {
      city: `Amsterdam`,
      offer: HOTELS[0],
    },
    [NameSpace.USER]: {
      authorizationStatus: AuthorizationStatus.NO_AUTH,
      authInfo: {},
    }
  });

  const tree = renderer.create(
      <Provider store={store}>
        <Router history={history}>
          <DetailOffer
            onBookmarkClick={noop}
          />
        </Router>
      </Provider>
  ).toJSON;

  expect(tree).toMatchSnapshot();
});
