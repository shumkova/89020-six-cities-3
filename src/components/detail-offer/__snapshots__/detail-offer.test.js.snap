// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render DetailOffer 1`] = `[Function]`;
